server/app/services/cloud/aws/AwsPublicStorage.java

```diff
@@ -0,0 +1,97 @@
+package services.cloud.aws;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.typesafe.config.Config;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import play.Environment;
+import services.cloud.PublicStorageClient;
+import software.amazon.awssdk.regions.Region;
+
+/** An AWS Simple Storage Service (S3) implementation of public storage. */
+@Singleton
+public final class AwsPublicStorage implements PublicStorageClient {
+  private static final String AWS_PUBLIC_S3_BUCKET_CONF_PATH = "aws.s3.public_bucket";
+  private static final String AWS_PUBLIC_S3_FILE_LIMIT_CONF_PATH = "aws.s3.public_file_limit_mb";
+
+  private final AwsStorageUtils awsStorageUtils;
+  private final Region region;
+  private final Credentials credentials;
+  private final String bucket;
+  private final int fileLimitMb;
+  private final Client client;
+
+  @Inject
+  public AwsPublicStorage(
+      AwsStorageUtils awsStorageUtils,
+      AwsRegion region,
+      Credentials credentials,
+      Config config,
+      Environment environment) {
+    this.awsStorageUtils = checkNotNull(awsStorageUtils);
+    this.region = checkNotNull(region).get();
+    this.credentials = checkNotNull(credentials);
+    this.bucket = checkNotNull(config).getString(AWS_PUBLIC_S3_BUCKET_CONF_PATH);
+    this.fileLimitMb = checkNotNull(config).getInt(AWS_PUBLIC_S3_FILE_LIMIT_CONF_PATH);
+    if (environment.isDev()) {
+      client = new LocalStackClient(config, awsStorageUtils);
+    } else if (environment.isTest()) {
+      client = new NullClient();
+    } else {
+      client = new AwsClient();
```

Comment:
suggestion (non-blocking): what do you think about checking explicitly for `isProd` here and then adding an `else` statement that returns a null client? I don't know what other environments we might ever have, but it would protect against returning the actual client in that situation.

Reply:
That sounds great, done. I can update the applicant storage classes to do something similar in a separate PR.